src/js/containers/search/topFilterBar/TopFilterBarContainer.jsx

```diff
@@ -140,6 +146,30 @@ export class TopFilterBarContainer extends React.Component {
         return null;
     }
 
+
+    /**
+     * Logic for parsing the current Redux keyword filter into a JS object that can be parsed by the
+     * top filter bar
+     */
+    prepareKeywords(props) {
+        let selected = false;
+        const filter = {};
+
+        if (props.keyword) {
+        // keyword exists
+            selected = true;
+            filter.code = 'keyword';
+            filter.name = 'Keyword';
+
+            filter.values = props.keyword;
```

Comment:
TopFilterBar.jsx, line 40: ```const filters = this.props.filters.map((filter) => {
            filterCount += filter.values.length;``` // if this is a string, it'll add the length of the string
...
}
```